### PR TITLE
Open Banking Component

### PR DESCRIPTION
**Pull request by ozgur00**

### CODE REVIEW

File: paybybank/src/main/java/com/adyen/checkout/paybybank/PayByBankViewProvider.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+package com.adyen.checkout.paybybank
+
+import android.content.Context
+import android.util.AttributeSet
+import com.adyen.checkout.components.ui.ComponentView
+import com.adyen.checkout.components.ui.ViewProvider
+import com.adyen.checkout.components.ui.view.ComponentViewType
+
+object PayByBankViewProvider: ViewProvider {
```
Comment (OscarSpruit) [inline paybybank/src/main/java/com/adyen/checkout/paybybank/PayByBankViewProvider.kt:17]: Can be `internal`

File: paybybank/src/main/java/com/adyen/checkout/paybybank/DefaultPayByBankDelegate.kt [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            } else {
+                null
+            }
+        }
+
+    private fun List<InputDetail>?.getLegacyIssuers(): List<IssuerModel> =
+        this.orEmpty()
+            .flatMap { it.items.orEmpty() }
+            .mapNotNull { item ->
+                val (id, name) = item
```
Comment (OscarSpruit) [inline paybybank/src/main/java/com/adyen/checkout/paybybank/DefaultPayByBankDelegate.kt:128]: I believe you can already do destructuring in the line above:
```Kotlin
.mapNotNull { (id, name) -> 
```

File: paybybank/src/main/java/com/adyen/checkout/paybybank/DefaultPayByBankDelegate.kt [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        )
+    }
+
+    override fun getIssuers(): List<IssuerModel> {
+        return paymentMethod.issuers?.mapToModel() ?: paymentMethod.details.getLegacyIssuers()
+    }
+
+    private fun List<Issuer>.mapToModel(): List<IssuerModel> =
+        this.mapNotNull { issuer ->
+            val (id, name, isDisabled) = issuer
```
Comment (OscarSpruit) [inline paybybank/src/main/java/com/adyen/checkout/paybybank/DefaultPayByBankDelegate.kt:116]: Same here

File: paybybank/proguard-rules.pro [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Add project specific ProGuard rules here.
```
Comment (OscarSpruit) [inline paybybank/proguard-rules.pro:1]: We don't really need these files if there are no rules in them. For now we can keep them for consistency, but it looks a bit cleaner if all the comments here are removed

File: paybybank/src/main/java/com/adyen/checkout/paybybank/DefaultPayByBankDelegate.kt (hunk 2 above)
Comment (ozgur00) [reply paybybank/src/main/java/com/adyen/checkout/paybybank/DefaultPayByBankDelegate.kt:128]: Tried it but didn't work.

**Review by OscarSpruit — APPROVED** [review]

**Pull request merged**